pandas/tseries/base.py

```diff
@@ -591,6 +595,8 @@ def __sub__(self, other):
                 return self.shift(-other)
             elif isinstance(other, (tslib.Timestamp, datetime)):
                 return self._sub_datelike(other)
+            elif isinstance(other, _period.Period):
```

Comment:
``NaT`` will be handled ``_sub_datelike``, currently raises ``AbstractMethodError``.

As in #12759, ``Period`` doesn't support ``NaT`` and use it's own ``Period(NaT) ATM.

```
pd.Period('2011-01', freq='M') - pd.NaT
# NaT

pd.NaT - pd.Period('2011-01', freq='M')
# AttributeError: 'NaTType' object has no attribute 'freq'
```